### DISCUSSION TITLE
How to get other column value in ImportColumn resolveUsing?

### DISCUSSION THREAD

**Discussion by kpsanmao** [filamentphp/filament#15058] [Help]
### Package

Actions

### Package Version

v3.2.128

### How can we help you?

For example, I want to import：student_id，grade, but in my csv file, student_id is not convenient, I want to get student_id by querying the database by student name and other fields like gender or date of birth in csv during import. in the manual, I found that I can customize the parsing through resolveUsing method procedure, but the example in the manual only contains the values of the current column, in my problem I need to get the values of other columns and retrieve them by combination, how should I sit?
```
            ImportColumn::make('student')
                ->relationship(resolveUsing: function (string $state){
                    return Student::query()
                        ->where('name', $state)
                        ->where('gender', $gender)//how to get gender column in csv
                        ->first();
                })
                ->rules(['required', 'max:255']),
```

**Comment by tonypartridge** [comment]
You can pass in $data within the function and you should be able to pass in $rawData for the pre-casted data. you ca Log to see the data to when testing. 


```php

use Illuminate\Support\Facades\Log;

     ImportColumn::make('student')
            ->relationship(resolveUsing: function (string $state, $data, $rawData): ?Author {
                    Log::info('Data:', ['Data' => $data, 'RawData' => $rawData]);

                    return Student::query()
                        ->where('name', $state)
                        ->where('gender', $gender)//how to get gender column in csv
                        ->first();
    })
```
